src/_stories/execute/utilities.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+from _stories.exceptions import StoryDefinitionError
+from _stories.returned import Failure
+from _stories.returned import Result
+from _stories.returned import Skip
+from _stories.returned import Success
+
+
+def validate_result_type(method, result_type, result):
+    if hasattr(method, "story_name"):
+        method_name = "{}.{}".format(method.story_name, method.__name__)
+    else:
+        try:
+            method_name = method.__qualname__
+        except AttributeError:
+            # Python 2 fallback
+            method_name = "{}.{}".format(
+                method.im_self.__class__.__name__, method.__name__
+            )
+    if result_type not in (Result, Success, Failure, Skip):
+        if result is True:
```

Comment:
I don't understand why do we need such heuristic. Why not raise the same message for any kind of result?

Reply:
Because it makes sense to return True or False to mark success or failure (respectively).
Returning None without an exception is also a successful execution in most Python code.
That's why it's ambiguous.
A lot of libraries use these properties to determine success or failure.